lib/IRGen/IRGen.cpp

```diff
@@ -490,10 +490,13 @@ static createTargetMachine(IRGenOptions &Opts, ASTContext &Ctx) {
   }
 
   // Create a target machine.
-  llvm::TargetMachine *TargetMachine
-    = Target->createTargetMachine(Triple.str(), CPU,
-                                  targetFeatures, TargetOpts, Reloc::PIC_,
-                                  CodeModel::Default, OptLevel);
+  auto cmodel = CodeModel::Default;
+  if (Triple.isWindowsCygwinEnvironment())
+    cmodel = CodeModel::Large;
```

Comment:
Interesting.  Thats something that I somehow did not hit.  However, even for the MSVC environment, we should be using `LARGEADDRESSAWARE` mode.  So, perhaps its best to just always enable the large code model.  I think the selection can just be inlined via a ternary.